go/consensus/api/transaction/transaction.go

```diff
@@ -148,64 +149,75 @@ type PrettyTransaction struct {
 
 // SignedTransaction is a signed transaction.
 type SignedTransaction struct {
-	signature.Signed
+	multisig.Envelope
 }
 
 // Hash returns the cryptographic hash of the encoded transaction.
 func (s *SignedTransaction) Hash() hash.Hash {
 	return hash.NewFrom(s)
 }
 
+// Open first verifies the blob signature and then unmarshals the blob.
+func (s *SignedTransaction) Open(tx *Transaction) error { // nolint: interfacer
+	return s.Envelope.Open(SignatureContext, tx)
+}
+
+// PrettyType returns a representation of the type that can be used for pretty printing.
+func (s SignedTransaction) PrettyType() (interface{}, error) {
+	var tx Transaction
+	if err := cbor.Unmarshal(s.Envelope.Payload, &tx); err != nil {
+		return nil, fmt.Errorf("malformed signed payload: %w", err)
+	}
+	return multisig.NewPrettyEnvelope(s.Envelope, tx)
+}
+
+// SingleSign serializes the Transaction and signs the result.
+//
+// Note: This is a convenience routine that does not support runtimes
+// backed by accounts with more than 1 signer.
+func SingleSign(signer signature.Signer, account *multisig.Account, tx *Transaction) (*SignedTransaction, error) {
+	if len(account.Signers) != 1 {
+		return nil, fmt.Errorf("attemtped to single-sign multi-sig transaction")
+	}
+	rawTx := cbor.Marshal(tx)
+	txSig, err := multisig.Sign(signer, account, SignatureContext, rawTx)
+	if err != nil {
+		return nil, err
+	}
+	envelope, err := multisig.NewEnvelope(account, []*signature.Signature{txSig}, rawTx)
+	if err != nil {
+		return nil, err
+	}
+	return &SignedTransaction{*envelope}, nil
+}
+
 // PrettyPrint writes a pretty-printed representation of the type
 // to the given writer.
 func (s SignedTransaction) PrettyPrint(prefix string, w io.Writer) {
 	fmt.Fprintf(w, "%sHash: %s\n", prefix, s.Hash())
 
-	fmt.Fprintf(w, "%sSigner: %s\n", prefix, s.Signature.PublicKey)
-	fmt.Fprintf(w, "%s        (signature: %s)\n", prefix, s.Signature.Signature)
+	// TODO: Someone that cares probably can figure out a better way
```

Comment:
Why wouldn't this just show the stringified staking address derived from the account?

Reply:
From `go/staking/api.go`: `"github.com/oasisprotocol/oasis-core/go/consensus/api/transaction"`